library/proc_macro/src/quote.rs

```diff
@@ -477,6 +407,381 @@ fn collect_meta_vars(content_stream: TokenStream) -> Vec<Ident> {
+        {
+            true
+        }
+        [TokenTree::Punct(t1), TokenTree::Punct(t2)]
+            if t1.as_char() == '-' && t1.spacing() == Spacing::Joint && t2.as_char() == '>' =>
+        {
+            true
+        }
+        [TokenTree::Punct(t1), TokenTree::Punct(t2)]
+            if t1.as_char() == '<' && t1.spacing() == Spacing::Joint && t2.as_char() == '-' =>
+        {
+            true
+        }
+        [TokenTree::Punct(t1), TokenTree::Punct(t2)]
+        if t1.as_char() == '%' && t1.spacing() == Spacing::Joint && t2.as_char() == '=' =>
+        {
+            true
+        }
+        [TokenTree::Punct(t1), TokenTree::Punct(t2)]
+            if t1.as_char() == '=' && t1.spacing() == Spacing::Joint && t2.as_char() == '>' =>
+        {
+            true
+        }
+        [TokenTree::Punct(t1), TokenTree::Punct(t2)]
+            if t1.as_char() == '<' && t1.spacing() == Spacing::Joint && t2.as_char() == '<' =>
+        {
+            true
+        }
+        [TokenTree::Punct(t1), TokenTree::Punct(t2)]
+            if t1.as_char() == '>' && t1.spacing() == Spacing::Joint && t2.as_char() == '>' =>
+        {
+            true
+        }
+        [TokenTree::Punct(t1), TokenTree::Punct(t2)]
+            if t1.as_char() == '-' && t1.spacing() == Spacing::Joint && t2.as_char() == '=' =>
+        {
+            true
+        }
+        [TokenTree::Punct(single_quote), TokenTree::Ident(_)] // lifetime
+            if single_quote.as_char() == '\'' && single_quote.spacing() == Spacing::Joint =>
+        {
+            true
+        }
+        [TokenTree::Punct(t1)] if t1.as_char() == '#' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == ',' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == '.' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == ';' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == ':' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == '+' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == '@' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == '!' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == '^' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == '&' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == '/' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == '=' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == '>' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == '<' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == '|' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == '?' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == '%' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == '*' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == '-' => true,
+        [TokenTree::Punct(t1)] if t1.as_char() == '_' => true,
+        [TokenTree::Ident(_)] => true,
+        [TokenTree::Group(_)] => true,
+        [_x] => true,
```

Comment:
This is incorrectly returning true for `TokenTree::Punct('\'', Spacing::Joint)`.

```suggestion
        [TokenTree::Ident(_) | TokenTree::Group(_) | TokenTree::Literal(_)] => true,
```